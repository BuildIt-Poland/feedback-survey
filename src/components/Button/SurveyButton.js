import React from 'react';
import styled from 'styled-components';
import { Button } from 'grommet';
import { string } from 'prop-types';

import { colorBlue, colorWhite, spacingLarge, spacingXLarge, fontMedium } from '../../styles/designTokens';
import media from '../../styles/media';

const StyledButton = styled(Button)`
  color: ${colorBlue};
  border-radius: 1rem;
  border-color: ${colorBlue};
  margin: 1rem auto;
  padding: 2rem 4rem;
  font-size: 1.8rem;

  :hover {
    background-color: ${colorBlue};
    color: ${colorWhite};
  }

  :focus {
    box-shadow: 0 0 0.2rem 0.2rem ${colorBlue};
  }

  ${media.tablet`
    padding: ${spacingLarge} ${spacingXLarge};
    font-size: ${fontMedium};
  `}
`;

const SurveyButton = ({ label, buttonType = 'button' }) => (
  <StyledButton label={label} color={colorBlue} type={buttonType} hoverIndicator="background" size="large" />
);

SurveyButton.protoTypes = {
  label: string.isRequired,
  buttonType: string
};

export default SurveyButton;
