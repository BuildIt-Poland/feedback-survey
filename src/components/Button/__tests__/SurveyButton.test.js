import React from 'react';
import { create } from 'react-test-renderer';

import SurveyButton from '../SurveyButton';

describe('COMPONENT - SurveyButton', () => {
  it('renders correctly', () => {
    const component = create(<SurveyButton label="test" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
