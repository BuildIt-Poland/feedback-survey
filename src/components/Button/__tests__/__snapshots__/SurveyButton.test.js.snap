// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SurveyButton renders correctly 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 eUfLDp"
>
  <button
    className="StyledButton-sc-323bzc-0 hlJKLl"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    size="large"
    type="button"
  >
    test
  </button>
</div>
`;
